docs/best-practices/retry-service-specific.md

```diff
@@ -423,33 +384,77 @@ However, in the current version of the Transient Fault Handling Application Bloc
 {
     GuardConnectionIsNotNull(command);
 
-    // Check if retry policy was specified, if not, use the default retry policy.
-    return await (cmdRetryPolicy ?? RetryPolicy.NoRetry).ExecuteAsync(async () =>
+    int eventualSuccesses = 0;
+    int retries = 0;
+    int eventualFailures = 0;
+
+    // Define our policy:
+    var policy = Policy.Handle<Exception>().WaitAndRetryAsync(
+        retryCount: 3, // Retry 3 times
+        sleepDurationProvider: attempt => TimeSpan.FromMilliseconds(200), // Wait 200ms between each try.
+        onRetry: (exception, calculatedWaitDuration) => // Capture some info for logging!
+        {
+            // This is your new exception handler! 
+            // Tell the user what they've won!
+            Console.ForegroundColor = ConsoleColor.Yellow;
+            Console.WriteLine($"Policy logging: {exception.Message}");
+            retries++;
+        });
+
+    int i = 0;
+
+    // Do the following until a key is pressed
+    while (!Console.KeyAvailable && !cancellationToken.IsCancellationRequested)
     {
-        var hasOpenConnection = await EnsureValidConnectionAsync(command, conRetryPolicy).ConfigureAwait(false);
+        i++;
 
         try
         {
-            return await command.ExecuteReaderAsync().ConfigureAwait(false);
-        }
-        catch (Exception)
-        {
-            if (hasOpenConnection && command.Connection != null && command.Connection.State == ConnectionState.Open)
+            // Retry the following call according to the policy - 3 times.
+            await policy.ExecuteAsync<SqlDataReader>(async token =>
             {
-                command.Connection.Close();
-            }
+                // This code is executed within the Policy 
+
+                // Make a request and get a response
+                if (conn.State != System.Data.ConnectionState.Open) await conn.OpenAsync();
+                var command = new SqlCommand("SELECT COUNT(*) FROM SomeTable", conn);
+
+                var reader = command.ExecuteReader(System.Data.CommandBehavior.Default);
+
+                eventualSuccesses++;
 
-            throw;
+                return reader;
+
+            }, cancellationToken);
+        }
+        catch (Exception e)
+        {
+            Console.ForegroundColor = ConsoleColor.Red;
+            Console.WriteLine($"Request {i} eventually failed with: {e.Message}");
+            eventualFailures++;
         }
-    }).ConfigureAwait(false);
+
+        // Wait half second
+        await Task.Delay(TimeSpan.FromSeconds(0.5), cancellationToken);
+    }
+
+    Console.WriteLine("");
+    Console.WriteLine($"Total requests made                 : {i}" + i);
+    Console.WriteLine($"Requests which eventually succeeded : {eventualSuccesses}");
+    Console.WriteLine($"Retries made to help achieve success: {retries}");
+    Console.WriteLine($"Requests which eventually failed    : {eventualFailures}");
 }
 ```
 
```

Comment:
Thank you for mentioning Polly in this guidance. We are glad simply if [Polly](https://github.com/App-vNext/Polly) is useful to the wider .NET community. I am offering some comments as (disclosure) [lead developer on the Polly library](https://github.com/App-vNext/Polly/graphs/contributors).

The code example for Polly with ADO.NET could be simplified, per code suggestion below:

```csharp
public async static Task<SqlDataReader> ExecuteReaderWithRetryAsync(this SqlCommand command)
{
    GuardConnectionIsNotNull(command);

    var policy = Policy.Handle<Exception>().WaitAndRetryAsync(
        retryCount: 3, // Retry 3 times
        sleepDurationProvider: attempt => TimeSpan.FromMilliseconds(200 * Math.Pow(2, attempt - 1)), // Exponential backoff based on an initial 200ms delay.
        onRetry: (exception, attempt) => 
        {
            // Capture some info for logging/telemetry.  
            logger.LogWarn($"ExecuteReaderWithRetryAsync: Retry {attempt} due to {exception}.");
        });

    // Retry the following call according to the policy.
    await policy.ExecuteAsync<SqlDataReader>(async token =>
    {
        // This code is executed within the Policy 

        if (conn.State != System.Data.ConnectionState.Open) await conn.OpenAsync(token);
        return await command.ExecuteReaderAsync(System.Data.CommandBehavior.Default, token);

    }, cancellationToken);
}
```

Changes from the original are:

+ Remove material around `Console.ReadKey()`, `Console.Writeline()`, which was only relevant to the [Polly-Samples demo](https://github.com/App-vNext/Polly-Samples) you originally referenced, and is not relevant in production.
+ Replace with example logging.
+ Change wait between retries to an exponential backoff example with Polly (following the recommendation you have for exponential backoff with SQL, earlier in the article)
+ Change all SQL calls to async calls taking advantage of the passed-in `CancellationToken`

Hope that this is useful.